src/core/mesh/qgsmeshtracerenderer.cpp

```diff
@@ -0,0 +1,605 @@
+
+  QgsRectangle layerExtent = mRenderContext.coordinateTransform().transform( mLayerExtent );
+  QgsRectangle mapExtent = mRenderContext.extent();
+
+  QgsRectangle interestZoneExtent = layerExtent.intersect( mapExtent );
+  if ( interestZoneExtent == QgsRectangle() )
+  {
+    mValid = false;
+    mFieldSize = QSize();
+    mFieldTopLeftInDeviceCoordinates = QPoint();
+    initField();
+    return;
+  }
+
+  QgsPointXY interestZoneTopLeft = deviceMapToPixel.transform( QgsPointXY( interestZoneExtent.xMinimum(), interestZoneExtent.yMaximum() ) );
+  QgsPointXY interestZoneBottomRight = deviceMapToPixel.transform( QgsPointXY( interestZoneExtent.xMaximum(), interestZoneExtent.yMinimum() ) );
+
+
+  mFieldTopLeftInDeviceCoordinates = interestZoneTopLeft.toQPointF().toPoint();
+  QPoint mFieldBottomRightInDeviceCoordinates = interestZoneBottomRight.toQPointF().toPoint();
+  int fieldWidthInDeviceCoordinate = mFieldBottomRightInDeviceCoordinates.x() - mFieldTopLeftInDeviceCoordinates.x();
+  int fieldHeightInDeviceCoordinate = mFieldBottomRightInDeviceCoordinates.y() - mFieldTopLeftInDeviceCoordinates.y();
+
+
+  int fieldWidth = int( fieldWidthInDeviceCoordinate / mFieldResolution );
+  int fieldHeight = int( fieldHeightInDeviceCoordinate / mFieldResolution );
+
+
+  //increase the field size if this size is not adjusted to extent of zone of interest in device coordinates
+  if ( fieldWidthInDeviceCoordinate % mFieldResolution > 0 )
+    fieldWidth++;
+  if ( fieldHeightInDeviceCoordinate % mFieldResolution > 0 )
+    fieldHeight++;
+
+  mFieldSize.setWidth( fieldWidth );
+  mFieldSize.setHeight( fieldHeight );
+
+
+  double mapUnitperFieldPixel;
+  if ( interestZoneExtent.width() > 0 )
+    mapUnitperFieldPixel = interestZoneExtent.width() / fieldWidthInDeviceCoordinate * mFieldResolution;
+  else
+    mapUnitperFieldPixel = 1e-8;
+
+  int fieldRightDevice = mFieldTopLeftInDeviceCoordinates.x() + mFieldSize.width() * mFieldResolution;
+  int fieldBottomDevice = mFieldTopLeftInDeviceCoordinates.y() + mFieldSize.height() * mFieldResolution;
+  QgsPointXY fieldRightBottomMap = deviceMapToPixel.toMapCoordinates( fieldRightDevice, fieldBottomDevice );
+
+  int fieldTopDevice = mFieldTopLeftInDeviceCoordinates.x();
+  int fieldLeftDevice = mFieldTopLeftInDeviceCoordinates.y();
+  QgsPointXY fieldTopLeftMap = deviceMapToPixel.toMapCoordinates( fieldTopDevice, fieldLeftDevice );
+
+  double xc = ( fieldRightBottomMap.x() + fieldTopLeftMap.x() ) / 2;
+  double yc = ( fieldTopLeftMap.y() + fieldRightBottomMap.y() ) / 2;
+
+  mMapToFieldPixel = QgsMapToPixel( mapUnitperFieldPixel,
+                                    xc,
+                                    yc,
+                                    fieldWidth,
+                                    fieldHeight, 0 );
+
+  initField();
+  mValid = true;
+}
+
+void QgsMeshTraceField::addTrace( QgsPointXY startPoint )
+{
+  addTrace( mMapToFieldPixel.transform( startPoint ).toQPointF().toPoint() );
```

Comment:
transform car raise exception

Reply:
QgsMapToPixel too ? Not sure, but it seems not.